code/datums/components/slippery.dm

```diff
@@ -1,17 +1,27 @@
+/// Slippery component, for making anything slippery. Of course.
 /datum/component/slippery
+	/// If the slip forces you to drop held items.
 	var/force_drop_items = FALSE
+	/// How long the slip keeps you knocked down.
 	var/knockdown_time = 0
+	/// How long the slip paralyzes for.
 	var/paralyze_time = 0
+	/// Flags for how slippery the parent is.
 	var/lube_flags
+	/// A proc callback to call on slip.
 	var/datum/callback/callback
+	/// If parent is an item, this is the person currently holding/wearing the parent (or the parent if no one is holding it)
 	var/mob/living/holder
+	/// Whitelist of item slots the parent can be equipped in that make the holder slippery. If null or empty, it will always make the holder slippery.
+	var/list/slot_whitelist
 
-/datum/component/slippery/Initialize(_knockdown, _lube_flags = NONE, datum/callback/_callback, _paralyze, _force_drop = FALSE)
+/datum/component/slippery/Initialize(_knockdown, _lube_flags = NONE, datum/callback/_callback, _paralyze, _force_drop = FALSE, _slot_whitelist)
```

Comment:
While you're at it, change `_knockdown`, `_lube_flags`, etc to `knockdown`, `lube_flags`, and so forth, and then do:

```dm
src.slot_whitelist = slot_whitelist
```

The purpose of this is so that when you use named arguments (which you do later on), you don't expose the implementation detail implied by the `_name` style (the implementation detail being "we have internal variables that are named the same").